solcast/historic.py

```diff
@@ -3,49 +3,81 @@
 
 
 def radiation_and_weather(
-    latitude: float, longitude: float, start: str, **kwargs
+    latitude: float,
+    longitude: float,
+    start: str,
+    end: str = None,
+    duration: str = None,
+    **kwargs,
 ) -> Response:
     """
     Get historical irradiance and weather estimated actuals for up to 31 days of data
     at a time for a requested location, derived from satellite (clouds and irradiance
     over non-polar continental areas) and numerical weather models (other data).
     Data is available from 2007-01-01T00:00Z up to real time estimated actuals.
 
+    To define the length of the period you can use one of "end" or "duration" parameter.
+
     See https://docs.solcast.com.au/ for full list of parameters.
     """
 
     client = Client(base_url=base_url, endpoint=historic_radiation_and_weather)
 
-    return client.get(
-        {
-            "latitude": latitude,
-            "longitude": longitude,
-            "start": start,
-            "format": "json",
-            **kwargs,
-        }
-    )
+    assert (end is None and duration is not None) | (

```

Comment:
`ValueError` more appropriate for a library.

```python
if end and duration:
     raise ValueError("only supply one of duration or end")
 ```
BUT

https://api.solcast.com.au/data/historic/radiation_and_weather?latitude=39.44&longitude=-121.45&start=2023-09-12&end=2023-09-13&duration=PT24H&format=json

already does this:
![image](https://github.com/Solcast/solcast-api-python-sdk/assets/135086092/85e1e888-4e63-47c4-b192-ececeddfa334)

So:

remove this altogether, keep the pattern of relying on API to do error messages.
If we decide to move validation to the SDK in the future, let's worry about it then.
Although this validation is harmless, it sets the expectation for putting validation in the SDK - BUT we don't have a good idea about what that validation should look like, so we risk a mish-mash of rules.